docs/sql-ref-syntax-ddl-alter-table.md

```diff
@@ -83,6 +83,37 @@ ALTER TABLE table_identifier ADD COLUMNS ( col_spec [ , col_spec ... ] )
 </dl>
 
 
+### ADD PARTITION
+
+`ALTER TABLE ADD PARTITION` adds a new partition to an existing, partitioned table.
+
+#### Syntax
+
+```sql
+ALTER TABLE <table_identifier> ADD [IF NOT EXISTS]
+  PARTITION <partition_spec> [LOCATION 'location']
+  [ PARTITION <partition_spec> [LOCATION 'location'] ...]
+;
+
+<table_identifier> ::=
+  [ database_name. ]table_name
+
+<partition_spec> ::=
+  (partition_column = partition_col_value, partition_column = partition_col_value, ...)
+```
+
+#### Examples
```

Comment:
Instead of having all examples at the end, I think it's clearer and easier to follow when each command has examples in its own section.